functions/brinks/src/middleware/decodeMessage.js

```diff
@@ -0,0 +1,65 @@
+import debug from 'debug';
+import * as Sentry from '@sentry/node';
+import { decodeMessage } from '../gcp/pubsub';
+
+const dlog = debug('that:api:functions:brinks:decodeMessageMw');
+
+export default function decodePubSubMessage(req, res, next) {
+  dlog('decodeMessage middleware called');
+
+  const thatBrinks = {
+    thatApp: 'brinks',
+    isProcessed: false,
+    decodeMessage: true,
+    sentryLevel: '',
+    stages: ['decodeMessage'],
+  };
+  req.thatBrinks = thatBrinks;
+
+  const { data, messageId, attributes } = req.body.message;
+  if (!data || !messageId || !attributes) {
+    thatBrinks.errorMsg = `Invalid PubSub message received, missing properties`;
+    thatBrinks.sentryLevel = 'error';
+    console.log(thatBrinks.errorMsg);
+    res.status(400);
+    return next(thatBrinks.errorMsg);
+  }
+
+  const decodedData = decodeMessage(data);
+  if (!decodedData) {
+    thatBrinks.errorMsg = `Unable to decode message data`;
+    thatBrinks.sentryLevel = 'error';
+    console.log(thatBrinks.errorMsg);
+    res.status(400);
+    return next(thatBrinks.errorMsg);
+  }
+
+  req.stripeEvent = decodedData.data;
+  req.pubSubMessage = {
+    data: decodedData,
+    messageId,
+    attributes,
+  };
+  thatBrinks.stripeEventType = decodedData.data.type;
+  thatBrinks.stripeEventId = decodedData.data.id;
+  thatBrinks.orderCreatedAt = new Date(decodedData.data.created * 1000);
+  Sentry.setTags({
+    stripeEventType: decodedData.data.type,
+    stripeEventId: decodedData.data.id,
+  });
+
+  return next();
+}
+
+/*
```

Comment:
Do we want to cut this? should it be in source?

Reply:
It is a scrubbed reference to what is received from PubSub. It is a good reference when we haven't seen this code for a long time. If you don't like it there we can remove it.